test/unit_tests/test_ahs_device.py

```diff
@@ -0,0 +1,518 @@
+    status: Status
+    pre_sequence: np.array
+    post_sequence: np.array
+
+
+DUMMY_RESULTS = [(DummyMeasurementResult(Status('Success'), np.array([1]), np.array([1])), np.array([0])),
+           (DummyMeasurementResult(Status('Success'), np.array([1]), np.array([0])), np.array([1])),
+           (DummyMeasurementResult(Status('Success'), np.array([0]), np.array([0])), np.array([np.NaN])),
+           (DummyMeasurementResult(Status('Failure'), np.array([1]), np.array([1])), np.array([np.NaN])),
+           (DummyMeasurementResult(Status('Success'), np.array([1, 1, 0]), np.array([1, 0, 0])), np.array([0, 1, np.NaN])),
+           (DummyMeasurementResult(Status('Success'), np.array([1, 1]), np.array([0, 0])), np.array([1, 1])),
+           (DummyMeasurementResult(Status('Success'), np.array([0, 1]), np.array([0, 0])), np.array([np.NaN, 1])),
+           (DummyMeasurementResult(Status('Failure'), np.array([1, 1]), np.array([1, 1])), np.array([np.NaN, np.NaN]))
+           ]
+
+
+
+class TestBraketAhsDevice:
+    """Tests that behaviour defined for both the LocalSimulator and the
+    Aquila hardware in the base device work as expected"""
+
+    @pytest.mark.parametrize("shots", [2, 1003])
+    @pytest.mark.parametrize("dev_cls, name, short_name", DEV_ATTRIBUTES)
+    def test_initialization(self, dev_cls, name, short_name, shots):
+        """Test the device initializes with the expected attributes"""
+
+        dev = dev_cls(wires=3, shots=shots)
+
+        assert dev._device.name == name
+        assert dev.short_name == short_name
+        assert dev.shots == shots
+        assert dev.ahs_program is None
+        assert dev.samples is None
+        assert dev.pennylane_requires == ">=0.29.0"
+        assert dev.operations == {"ParametrizedEvolution"}
+
+    def test_settings(self):
+        dev = dev_sim
+        assert isinstance(dev.settings, dict)
+        assert 'interaction_coeff' in dev.settings.keys()
+        assert len(dev.settings.keys()) == 1
+        assert dev.settings['interaction_coeff'] == 862690
+
+    @pytest.mark.parametrize("dev_cls, shots", [(BraketAquilaDevice, 1000),
+                                                (BraketAquilaDevice, 2),
+                                                (BraketLocalAquilaDevice, 1000),
+                                                (BraketLocalAquilaDevice, 2)])
+    def test_setting_shots(self, dev_cls, shots):
+        """Test that setting shots changes number of shots from default (100)"""
+        dev = dev_cls(wires=3, shots=shots)
+        assert dev.shots == shots
+
+    @pytest.mark.parametrize("shots", [0, None])
+    def test_no_shots_raises_error(self, shots):
+        """Test that an error is raised if shots are set to 0 or None"""
+        with pytest.raises(RuntimeError, match="This device requires shots"):
+            BraketLocalAquilaDevice(wires=3, shots=shots)
+
+    @pytest.mark.parametrize("dev_cls, wires", [(BraketAquilaDevice, 2),
```

Comment:
Is it alright to have tests that connect to hardware (without running a task) in unit testing, or does this only belong in integration testing?

Reply:
Unit tests should not connect to the service, and instead mock the service calls

Reply:
Could you point me to an example somewhere else in the tests where this is done? I'm not quite sure how to approach it!

Reply:
For example, take a look at [this test](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/test/unit_tests/test_braket_device.py#L1412-L1437)

Reply:
Thanks - I ended up using `monkeypatch` to do the tests for hardware, I spent some time looking at how the tests were done for the existing devices but had trouble transferring it over to the AHS device. Would you say the tests seem alright at this point, or does more work need to be done?